src/RigidBody/Contacts.cpp

```diff
@@ -45,6 +49,23 @@ unsigned int rigidbody::Contacts::AddConstraint(
     double acc)
 {
     ++*m_nbreConstraint;
+
+    // Check world_normal points to what axis
+    utils::String axis = "";
+    SCALAR_TO_DOUBLE(has_x, world_normal[0]);
+    SCALAR_TO_DOUBLE(has_y, world_normal[1]);
+    SCALAR_TO_DOUBLE(has_z, world_normal[2]);
+    if (has_x != 0){
+        axis += "x";
+    }
+    if (has_y != 0){
+        axis += "y";
+    }
+    if (has_z != 0){
+        axis += "z";
+    }
+
+    m_rigidContacts->push_back(NodeSegment(body_point, name, "",true,false, axis.c_str(),body_id));
```

Comment:
This is the best I could do... but I am not sure this is okay, could you add a test to validate that when you call this function it actually adds the axis properly?

Reply:
this works